DiagnosticsExtension/Models/ConnectionStringValidator/Exceptions/ConnectionStringResponseUtility.cs

```diff
@@ -0,0 +1,128 @@
+        public const string Credential = "__credential";
+        public const string ServiceUriMissed = "ServiceUriMissed";
+        public const string ValidCredentialValue = "managedidentity";
+        public const string FullyQualifiedNamespace = "__fullyQualifiedNamespace";
+        public const string ManagedIdentityCredentialMissing = "ManagedIdentityCredentialMissing";
+        #endregion
+
+        public static ConnectionStringValidationResult EvaluateResponseStatus(Exception e, ConnectionStringType type, ConnectionStringValidationResult.ManagedIdentityType identityType)
+        {
+            var response = new ConnectionStringValidationResult(type);
+            if (e is MalformedConnectionStringException)
+            {
+                response.Status = ConnectionStringValidationResult.ResultStatus.MalformedConnectionString;
+            }
+            else if (e.Message.Contains("managedIdentityCredentialMissing"))
+            {
+                response.Status = ConnectionStringValidationResult.ResultStatus.ManagedIdentityCredentialMissing;
+            }
+            else if (e.Message.Contains("Unauthorized") || e.Message.Contains("AuthorizationPermissionMismatch"))
+            {
+                if (identityType == ConnectionStringValidationResult.ManagedIdentityType.User)
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.UserAssignedManagedIdentity;
+                }
+                else
+                {
+                    response.Status = ConnectionStringValidationResult.ResultStatus.SystemAssignedManagedIdentity;
+                }
+            }
+            else if (e.Message.Contains("ManagedIdentityCredential"))
+            {
+                response.Status = ConnectionStringValidationResult.ResultStatus.ManagedIdentityCredential;
+            }
+            else if (e.Message.Contains("fullyQualifiedNamespace"))
+            {
+                response.Status = ConnectionStringValidationResult.ResultStatus.FullyQualifiedNamespaceMissed;
+            }
+            else if (e.Message.Contains("ServiceUriMissed"))
+            {
+                response.Status = ConnectionStringValidationResult.ResultStatus.ServiceUriMissed;
+            }
+            else if (e is EmptyConnectionStringException)
+            {
+                response.Status = ConnectionStringValidationResult.ResultStatus.EmptyConnectionString;
+            }
+            else if (e.InnerException != null &&
+                     e.InnerException.Message.Contains("The remote name could not be resolved"))
+            {
+                response.Status = ConnectionStringValidationResult.ResultStatus.DnsLookupFailed;
+            }
+            else if (e.InnerException != null && e.InnerException.InnerException != null &&
+                        e.InnerException.InnerException.Message.Contains("The remote name could not be resolved"))
```

Comment:
this looks like a duplicate of the previous else if block - combine them by removing the one above as this checks more exhaustively (e.g. e.InnerException.InnerException)